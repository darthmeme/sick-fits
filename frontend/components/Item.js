import React, { Component } from 'react'
import PropTypes from 'prop-types'
import Title from './styles/Title'
import ItemStyles from './styles/ItemStyles'
import PriceTag from './styles/PriceTag'
import Link from 'next/link'
import formatMoney from '../lib/formatMoney'
import DeleteItem from './DeleteItem'

export default class Item extends Component {
  static propTypes = {
    item: PropTypes.object.isRequired
  }
  
  render () {
    const { item } = this.props

    return (
      <ItemStyles>
        {item.image && <img src={item.image} />}
        <Title>
          <Link href={{ pathname: '/item', query: { id: item.id } }}>
            <a>{item.name}</a>
          </Link>
        </Title>
        <p>{item.description}</p>
        <div>
          <Link href={{ pathname: '/update', query: { id: item.id } }}><a>Edit ✏️</a></Link>
          <button>Add to card</button>
          <DeleteItem id={item.id} />
        </div>
        <PriceTag>{formatMoney(item.price)}</PriceTag>
      </ItemStyles>
    )
  }
}
