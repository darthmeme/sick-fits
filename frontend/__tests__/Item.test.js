import Item from '../components/Item'
import { shallow } from 'enzyme'
import toJSON from 'enzyme-to-json'

const fakeItem = {
  id: '123',
  name: 'Cool Item',
  description: 'A cool item',
  price: 10000,
  image: 'dog.jpg',
  largeImage: 'large-dog.jpg'
}

describe('<Item />', () => {
  it('renders and matches the snapshot', () => {
    const wrapper = shallow(<Item item={fakeItem} />)

    expect(toJSON(wrapper)).toMatchSnapshot()
  })

  // it('renders and displays properly', () => {
  //   const wrapper = shallow(<Item item={fakeItem} />)
  //   const priceTag = wrapper.find('PriceTag')
  //   const image = wrapper.find('img')

  //   expect(priceTag.children().text()).toBe('$100')
  //   expect(wrapper.find('Title a').text()).toBe(fakeItem.name)
  //   expect(image.props().src).toBe(fakeItem.image)
  // })

  // it('renders out the buttons properly', () => {
  //   const wrapper = shallow(<Item item={fakeItem} />)
  //   const buttonList = wrapper.find('.buttonList')

  //   expect(buttonList.children()).toHaveLength(3)
  //   expect(buttonList.find('Link').exists()).toBe(true)
  //   expect(buttonList.find('DeleteItem').exists()).toBe(true)
  //   expect(buttonList.find('AddToCart').exists()).toBe(true)
  // })
})
