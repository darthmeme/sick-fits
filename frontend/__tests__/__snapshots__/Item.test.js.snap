// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Item /> renders and matches the snapshot 1`] = `
<ItemStyles__Item>
  <img
    src="dog.jpg"
  />
  <Title>
    <Link
      href={
        Object {
          "pathname": "/item",
          "query": Object {
            "id": "123",
          },
        }
      }
    >
      <a>
        Cool Item
      </a>
    </Link>
  </Title>
  <p>
    A cool item
  </p>
  <div
    className="buttonList"
  >
    <Link
      href={
        Object {
          "pathname": "/update",
          "query": Object {
            "id": "123",
          },
        }
      }
    >
      <a>
        Edit ✏️
      </a>
    </Link>
    <AddToCart
      id="123"
    />
    <DeleteItem
      id="123"
    />
  </div>
  <PriceTag>
    $100
  </PriceTag>
</ItemStyles__Item>
`;
